### ISSUE TITLE
setsockopt(11, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket) with inotify

### ISSUE THREAD

**Issue by muahao** [balabit/syslog-ng#1292]
# syslog-ng
## Version of syslog-ng
```
#syslog-ng -V
syslog-ng 3.6.4
Installer-Version: 3.6.4
Revision:
Compile-Date: Dec 20 2016 19:29:39
Available-Modules: afuser,cryptofuncs,linux-kmsg-format,afsocket-tls,afprog,csvparser,confgen,pseudofile,system-source,affile,afsocket,sdjournal,dbparser,graphite,afamqp,afstomp,syslogformat,basicfuncs,afsocket-notls
Enable-Debug: off
Enable-GProf: off
Enable-Memtrace: off
Enable-IPv6: on
Enable-Spoof-Source: on
Enable-TCP-Wrapper: on
Enable-Linux-Caps: on
```

## Platform

rhel 7.2

# Issue
 I just write a hello.ko kernel module ,to printk kernel log messages, just want to test syslog-ng .  and I find when amount of kernel log messages print, strace shows ```setsockopt(11, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)```  Is this normal? Thanks a lot

## Failure

```
#gdb syslog-ng
GNU gdb (GDB) Red Hat Enterprise Linux 7.6.1-80.1.alios7
Copyright (C) 2013 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-redhat-linux-gnu".
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>...
Reading symbols from /usr/sbin/syslog-ng...Reading symbols from /usr/sbin/syslog-ng...(no debugging symbols found)...done.
(no debugging symbols found)...done.
Missing separate debuginfos, use: debuginfo-install syslog-ng-3.6.4-7.alios7.x86_64
(gdb) run
Starting program: /sbin/syslog-ng
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
Detaching after fork from child process 52988.
[Inferior 1 (process 52984) exited normally]
```

## Steps to reproduce
write a kernel module hello.ko

```
#cat hello.c
#include <linux/init.h>
#include <linux/module.h>
#include <linux/kthread.h>
#include <linux/delay.h>
#include <linux/err.h>

MODULE_LICENSE("Dual BSD/GPL");

static struct task_struct *tsk;

static int thread_function(void *data)
{
    int time_count = 0;
    do {
        printk(KERN_INFO "thread_function: %d times", ++time_count);
        msleep(50);
    }while(!kthread_should_stop());
    return time_count;
}

static int hello_init(void)
{
    printk(KERN_INFO "Hello, world!\n");

    tsk = kthread_run(thread_function, NULL, "mythread%d", 1);
    if (IS_ERR(tsk)) {
        printk(KERN_INFO "create kthread failed!\n");
    }
    else {
        printk(KERN_INFO "create ktrhead ok!\n");
    }
    return 0;
}

static void hello_exit(void)
{
    printk(KERN_INFO "Hello, exit!\n");
	if (IS_ERR(tsk)){
		printk(KERN_INFO "thread function already dispear");
    }else {
		printk(KERN_INFO "thread function is there,let's use kthread_stop to stop it\n");
        int ret = kthread_stop(tsk);
        printk(KERN_INFO "stop over,thread function has run %ds\n", ret);
	}
}

module_init(hello_init);
module_exit(hello_exit);

```

```
[root@rt2m09617.sqa.tbc /home/ahao.mah/main]
#ps axu | grep syslog-ng | grep -v grep
root      3826  0.0  0.0 138672 14736 ?        Ss   Dec23   4:16 /usr/sbin/syslog-ng -F -p /var/run/syslog-ng.pid

```

```
setsockopt(11, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
setsockopt(12, SOL_SOCKET, SO_OOBINLINE, [1], 4) = 0
```

```
[root@rt2m09617.sqa.tbc /home/ahao.mah/main]
#lsof -p 3826
COMMAND    PID USER   FD      TYPE             DEVICE  SIZE/OFF    NODE NAME
syslog-ng 3826 root  cwd       DIR                8,2      4096       2 /
syslog-ng 3826 root  rtd       DIR                8,2      4096       2 /
syslog-ng 3826 root  txt       REG                8,2     15944 2109245 /usr/sbin/syslog-ng
syslog-ng 3826 root  mem       REG                8,2   8388608  788256 /var/log/journal/4d36c13c14a94413a63bddda6df627e1/user-120063.journal
syslog-ng 3826 root  mem       REG                8,2   8388608  788254 /var/log/journal/4d36c13c14a94413a63bddda6df627e1/user-122575.journal
syslog-ng 3826 root  mem       REG                8,2  16777216  788242 /var/log/journal/4d36c13c14a94413a63bddda6df627e1/system.journal
syslog-ng 3826 root  mem       REG                8,2     79304 2233417 /usr/lib64/syslog-ng/libaffile.so
syslog-ng 3826 root  mem       REG                8,2     42520 2230726 /usr/lib64/libwrap.so.0.7.6
syslog-ng 3826 root  mem       REG                8,2     98736 2233365 /usr/lib64/libnet.so.1.7.0
syslog-ng 3826 root  mem       REG                8,2    122584 2233615 /usr/lib64/syslog-ng/libafsocket-tls.so
syslog-ng 3826 root  mem       REG                8,2     53568 2236532 /usr/lib64/syslog-ng/libsdjournal.so
syslog-ng 3826 root  mem       REG                8,2     61928 2230234 /usr/lib64/libnss_files-2.17.so
syslog-ng 3826 root  mem       REG                8,2     15504 2236534 /usr/lib64/syslog-ng/libsystem-source.so
syslog-ng 3826 root  mem       REG                8,2     11192 2236525 /usr/lib64/syslog-ng/libconfgen.so
syslog-ng 3826 root  mem       REG                8,2     15688 2230768 /usr/lib64/libkeyutils.so.1.5
syslog-ng 3826 root  mem       REG                8,2     62720 2233117 /usr/lib64/libkrb5support.so.0.1
syslog-ng 3826 root  mem       REG                8,2     68192 2229700 /usr/lib64/libbz2.so.1.0.6
syslog-ng 3826 root  mem       REG                8,2     91496 2230265 /usr/lib64/libelf-0.163.so
syslog-ng 3826 root  mem       REG                8,2     90632 2230263 /usr/lib64/libz.so.1.2.7
syslog-ng 3826 root  mem       REG                8,2    202576 2233098 /usr/lib64/libk5crypto.so.3.1
syslog-ng 3826 root  mem       REG                8,2     15840 2230536 /usr/lib64/libcom_err.so.2.1
syslog-ng 3826 root  mem       REG                8,2    950496 2233113 /usr/lib64/libkrb5.so.3.3
syslog-ng 3826 root  mem       REG                8,2    316528 2233090 /usr/lib64/libgssapi_krb5.so.2.2
syslog-ng 3826 root  mem       REG                8,2    297464 2230666 /usr/lib64/libdw-0.163.so
syslog-ng 3826 root  mem       REG                8,2    110808 2230254 /usr/lib64/libresolv-2.17.so
syslog-ng 3826 root  mem       REG                8,2     19384 2230678 /usr/lib64/libgpg-error.so.0.10.0
syslog-ng 3826 root  mem       REG                8,2    534488 2230684 /usr/lib64/libgcrypt.so.11.8.2
syslog-ng 3826 root  mem       REG                8,2    153192 2230229 /usr/lib64/liblzma.so.5.0.99
syslog-ng 3826 root  mem       REG                8,2    147120 2230257 /usr/lib64/libselinux.so.1
syslog-ng 3826 root  mem       REG                8,2   1141560 2230186 /usr/lib64/libm-2.17.so
syslog-ng 3826 root  mem       REG                8,2     19888 2230564 /usr/lib64/libattr.so.1.1.0
syslog-ng 3826 root  mem       REG                8,2     88720 2229643 /usr/lib64/libgcc_s-4.8.5-20150702.so.1
syslog-ng 3826 root  mem       REG                8,2     20032 2230545 /usr/lib64/libuuid.so.1.3.0
syslog-ng 3826 root  mem       REG                8,2   2017176 2233128 /usr/lib64/libcrypto.so.1.0.1e
syslog-ng 3826 root  mem       REG                8,2    449880 2233132 /usr/lib64/libssl.so.1.0.1e
syslog-ng 3826 root  mem       REG                8,2   2107816 2230151 /usr/lib64/libc-2.17.so
syslog-ng 3826 root  mem       REG                8,2    142304 2230250 /usr/lib64/libpthread-2.17.so
syslog-ng 3826 root  mem       REG                8,2     19520 2230182 /usr/lib64/libdl-2.17.so
syslog-ng 3826 root  mem       REG                8,2    398272 2230235 /usr/lib64/libpcre.so.1.2.0
syslog-ng 3826 root  mem       REG                8,2     20024 2230627 /usr/lib64/libcap.so.2.22
syslog-ng 3826 root  mem       REG                8,2     19936 2233353 /usr/lib64/libevtlog.so.0.0.0
syslog-ng 3826 root  mem       REG                8,2   1287896 2230700 /usr/lib64/libglib-2.0.so.0.4200.2
syslog-ng 3826 root  mem       REG                8,2      6928 2230714 /usr/lib64/libgthread-2.0.so.0.4200.2
syslog-ng 3826 root  mem       REG                8,2     15616 2230704 /usr/lib64/libgmodule-2.0.so.0.4200.2
syslog-ng 3826 root  mem       REG                8,2    113328 2230199 /usr/lib64/libnsl-2.17.so
syslog-ng 3826 root  mem       REG                8,2     44096 2230258 /usr/lib64/librt-2.17.so
syslog-ng 3826 root  mem       REG                8,2    676392 2233411 /usr/lib64/libsyslog-ng-3.6.so.0.0.0
syslog-ng 3826 root  mem       REG                8,2    164440 2229701 /usr/lib64/ld-2.17.so
syslog-ng 3826 root  mem       REG                8,2     16384  917946 /var/lib/syslog-ng/syslog-ng.persist
syslog-ng 3826 root  mem       REG                8,2    162560 2233099 /usr/lib64/libsystemd.so.0.6.0
syslog-ng 3826 root  mem       REG                8,2     26254 2230522 /usr/lib64/gconv/gconv-modules.cache
syslog-ng 3826 root    0r      CHR                1,3       0t0    1028 /dev/null
syslog-ng 3826 root    1u     unix 0xffff8817b51d3000       0t0  864917 socket
syslog-ng 3826 root    2u     unix 0xffff8817b51d3000       0t0  864917 socket
syslog-ng 3826 root    3u  a_inode                0,9         0    7269 [eventpoll]
syslog-ng 3826 root    4u  a_inode                0,9         0    7269 [eventfd]
syslog-ng 3826 root    5u     unix 0xffff8817b9d37400       0t0  835391 /var/lib/syslog-ng/syslog-ng.ctl
syslog-ng 3826 root    6u  a_inode                0,9         0    7269 [eventfd]
syslog-ng 3826 root    7u  a_inode                0,9         0    7269 [eventfd]
syslog-ng 3826 root    8u  a_inode                0,9         0    7269 [eventfd]
syslog-ng 3826 root    9u  a_inode                0,9         0    7269 [eventfd]
syslog-ng 3826 root   10u      REG                8,2     16384  917946 /var/lib/syslog-ng/syslog-ng.persist
syslog-ng 3826 root   11r  a_inode                0,9         0    7269 inotify
syslog-ng 3826 root   12u     IPv4            1087399       0t0     UDP rt2m09617.sqa.tbc:48464->10.101.0.55:syslog
syslog-ng 3826 root   13w      REG                8,2    657985  788208 /var/log/cron
syslog-ng 3826 root   14w      REG                8,2      5023  788232 /var/log/secure
syslog-ng 3826 root   15w      REG                8,2 138977768  788236 /var/log/kern
syslog-ng 3826 root   17r      REG                8,2  16777216  788242 /var/log/journal/4d36c13c14a94413a63bddda6df627e1/system.journal
syslog-ng 3826 root   19w      REG                8,2    220542  788214 /var/log/messages
syslog-ng 3826 root   20r      REG                8,2   8388608  788254 /var/log/journal/4d36c13c14a94413a63bddda6df627e1/user-122575.journal
syslog-ng 3826 root   22r      REG                8,2   8388608  788256 /var/log/journal/4d36c13c14a94413a63bddda6df627e1/user-120063.journal

```

```
# strace -p 3826
setsockopt(11, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
epoll_ctl(3, EPOLL_CTL_ADD, 11, {EPOLLIN, {u32=37500608, u64=37500608}}) = 0
epoll_wait(3, {{EPOLLIN, {u32=37411416, u64=37411416}}}, 8, 7545) = 1
read(4, "\1\0\0\0\0\0\0\0", 8)          = 8
writev(15, [{"Dec 26 14:27:27 rt2m09617.sqa.tb"..., 70}], 1) = 70
writev(15, [{"Dec 26 14:27:27 rt2m09617.sqa.tb"..., 70}], 1) = 70
writev(19, [{"Dec 26 14:27:27 rt2m09617.sqa.tb"..., 70}], 1) = 70
futex(0x23ace58, FUTEX_WAKE, 1)         = 0
writev(19, [{"Dec 26 14:27:27 rt2m09617.sqa.tb"..., 70}], 1) = 70
futex(0x23ace58, FUTEX_WAKE, 1)         = 0
epoll_ctl(3, EPOLL_CTL_MOD, 12, {EPOLLOUT, {u32=37487032, u64=37487032}}) = 0
epoll_wait(3, {{EPOLLOUT, {u32=37487032, u64=37487032}}}, 8, 7544) = 1
epoll_ctl(3, EPOLL_CTL_DEL, 12, {0, {u32=37487032, u64=37487032}}) = 0
sendto(12, "<6>Dec 26 14:27:27 rt2m09617.sqa"..., 73, 0, NULL, 0) = 73
sendto(12, "<6>Dec 26 14:27:27 rt2m09617.sqa"..., 73, 0, NULL, 0) = 73
fcntl(12, F_GETFD)                      = 0x1 (flags FD_CLOEXEC)
fcntl(12, F_GETFL)                      = 0x802 (flags O_RDWR|O_NONBLOCK)
setsockopt(12, SOL_SOCKET, SO_OOBINLINE, [1], 4) = 0
epoll_ctl(3, EPOLL_CTL_ADD, 12, {0, {u32=37487032, u64=37487032}}) = 0
epoll_wait(3, {{EPOLLIN, {u32=37500608, u64=37500608}}}, 8, 7544) = 1
read(11, "\2\0\0\0\2\0\0\0\0\0\0\0\20\0\0\0system.journal\0\0", 272) = 32
read(11, 0x7ffe6d0ec050, 272)           = -1 EAGAIN (Resource temporarily unavailable)
epoll_ctl(3, EPOLL_CTL_DEL, 11, {0, {u32=37500608, u64=37500608}}) = 0
write(4, "\1\0\0\0\0\0\0\0", 8)         = 8
fcntl(11, F_GETFD)                      = 0x1 (flags FD_CLOEXEC)
fcntl(11, F_GETFL)                      = 0x800 (flags O_RDONLY|O_NONBLOCK)
setsockopt(11, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
epoll_ctl(3, EPOLL_CTL_ADD, 11, {EPOLLIN, {u32=37500608, u64=37500608}}) = 0
epoll_wait(3, {{EPOLLIN, {u32=37411416, u64=37411416}}}, 8, 7412) = 1
read(4, "\1\0\0\0\0\0\0\0", 8)          = 8
writev(15, [{"Dec 26 14:27:27 rt2m09617.sqa.tb"..., 70}], 1) = 70
writev(15, [{"Dec 26 14:27:27 rt2m09617.sqa.tb"..., 70}], 1) = 70
writev(19, [{"Dec 26 14:27:27 rt2m09617.sqa.tb"..., 70}], 1) = 70
futex(0x23ace58, FUTEX_WAKE, 1)         = 0
writev(19, [{"Dec 26 14:27:27 rt2m09617.sqa.tb"..., 70}], 1) = 70
futex(0x23ace58, FUTEX_WAKE, 1)         = 0
epoll_ctl(3, EPOLL_CTL_MOD, 12, {EPOLLOUT, {u32=37487032, u64=37487032}}) = 0
epoll_wait(3, {{EPOLLOUT, {u32=37487032, u64=37487032}}}, 8, 7412) = 1
epoll_ctl(3, EPOLL_CTL_DEL, 12, {0, {u32=37487032, u64=37487032}}) = 0
sendto(12, "<6>Dec 26 14:27:27 rt2m09617.sqa"..., 73, 0, NULL, 0) = 73
sendto(12, "<6>Dec 26 14:27:27 rt2m09617.sqa"..., 73, 0, NULL, 0) = 73
fcntl(12, F_GETFD)                      = 0x1 (flags FD_CLOEXEC)
fcntl(12, F_GETFL)                      = 0x802 (flags O_RDWR|O_NONBLOCK)
setsockopt(12, SOL_SOCKET, SO_OOBINLINE, [1], 4) = 0
epoll_ctl(3, EPOLL_CTL_ADD, 12, {0, {u32=37487032, u64=37487032}}) = 0
epoll_wait(3, {{EPOLLIN, {u32=37500608, u64=37500608}}}, 8, 7411) = 1
read(11, "\2\0\0\0\2\0\0\0\0\0\0\0\20\0\0\0system.journal\0\0", 272) = 32
read(11, 0x7ffe6d0ec050, 272)           = -1 EAGAIN (Resource temporarily unavailable)
epoll_ctl(3, EPOLL_CTL_DEL, 11, {0, {u32=37500608, u64=37500608}}) = 0
write(4, "\1\0\0\0\0\0\0\0", 8)         = 8
fcntl(11, F_GETFD)                      = 0x1 (flags FD_CLOEXEC)
fcntl(11, F_GETFL)                      = 0x800 (flags O_RDONLY|O_NONBLOCK)
setsockopt(11, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
epoll_ctl(3, EPOLL_CTL_ADD, 11, {EPOLLIN, {u32=37500608, u64=37500608}}) = 0
epoll_wait(3, {{EPOLLIN, {u32=37411416, u64=37411416}}}, 8, 7277) = 1
read(4, "\1\0\0\0\0\0\0\0", 8)          = 8
writev(15, [{"Dec 26 14:27:27 rt2m09617.sqa.tb"..., 70}], 1) = 70
writev(15, [{"Dec 26 14:27:27 rt2m09617.sqa.tb"..., 70}], 1) = 70
writev(19, [{"Dec 26 14:27:27 rt2m09617.sqa.tb"..., 70}], 1) = 70
futex(0x23ace58, FUTEX_WAKE, 1)         = 0
writev(19, [{"Dec 26 14:27:27 rt2m09617.sqa.tb"..., 70}], 1) = 70
futex(0x23ace58, FUTEX_WAKE, 1)         = 0
epoll_ctl(3, EPOLL_CTL_MOD, 12, {EPOLLOUT, {u32=37487032, u64=37487032}}) = 0
epoll_wait(3, {{EPOLLOUT, {u32=37487032, u64=37487032}}}, 8, 7277) = 1
epoll_ctl(3, EPOLL_CTL_DEL, 12, {0, {u32=37487032, u64=37487032}}) = 0
sendto(12, "<6>Dec 26 14:27:27 rt2m09617.sqa"..., 73, 0, NULL, 0) = 73
sendto(12, "<6>Dec 26 14:27:27 rt2m09617.sqa"..., 73, 0, NULL, 0) = 73
fcntl(12, F_GETFD)                      = 0x1 (flags FD_CLOEXEC)
fcntl(12, F_GETFL)                      = 0x802 (flags O_RDWR|O_NONBLOCK)
setsockopt(12, SOL_SOCKET, SO_OOBINLINE, [1], 4) = 0
epoll_ctl(3, EPOLL_CTL_ADD, 12, {0, {u32=37487032, u64=37487032}}) = 0
epoll_wait(3, {{EPOLLIN, {u32=37500608, u64=37500608}}}, 8, 7276) = 1
read(11, "\2\0\0\0\2\0\0\0\0\0\0\0\20\0\0\0system.journal\0\0", 272) = 32
read(11, 0x7ffe6d0ec050, 272)           = -1 EAGAIN (Resource temporarily unavailable)
epoll_ctl(3, EPOLL_CTL_DEL, 11, {0, {u32=37500608, u64=37500608}}) = 0
write(4, "\1\0\0\0\0\0\0\0", 8)         = 8
fcntl(11, F_GETFD)                      = 0x1 (flags FD_CLOEXEC)
fcntl(11, F_GETFL)                      = 0x800 (flags O_RDONLY|O_NONBLOCK)
setsockopt(11, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
epoll_ctl(3, EPOLL_CTL_ADD, 11, {EPOLLIN, {u32=37500608, u64=37500608}}) = 0
epoll_wait(3, {{EPOLLIN, {u32=37411416, u64=37411416}}}, 8, 7145) = 1
read(4, "\1\0\0\0\0\0\0\0", 8)          = 8
writev(15, [{"Dec 26 14:27:27 rt2m09617.sqa.tb"..., 70}], 1) = 70
writev(15, [{"Dec 26 14:27:27 rt2m09617.sqa.tb"..., 70}], 1) = 70
writev(19, [{"Dec 26 14:27:27 rt2m09617.sqa.tb"..., 70}], 1) = 70
futex(0x23ace58, FUTEX_WAKE, 1)         = 0
writev(19, [{"Dec 26 14:27:27 rt2m09617.sqa.tb"..., 70}], 1) = 70
futex(0x23ace58, FUTEX_WAKE, 1)         = 0
epoll_ctl(3, EPOLL_CTL_MOD, 12, {EPOLLOUT, {u32=37487032, u64=37487032}}) = 0
epoll_wait(3, {{EPOLLOUT, {u32=37487032, u64=37487032}}}, 8, 7144) = 1
epoll_ctl(3, EPOLL_CTL_DEL, 12, {0, {u32=37487032, u64=37487032}}) = 0
sendto(12, "<6>Dec 26 14:27:27 rt2m09617.sqa"..., 73, 0, NULL, 0) = 73
sendto(12, "<6>Dec 26 14:27:27 rt2m09617.sqa"..., 73, 0, NULL, 0) = 73
fcntl(12, F_GETFD)                      = 0x1 (flags FD_CLOEXEC)
fcntl(12, F_GETFL)                      = 0x802 (flags O_RDWR|O_NONBLOCK)
setsockopt(12, SOL_SOCKET, SO_OOBINLINE, [1], 4) = 0
epoll_ctl(3, EPOLL_CTL_ADD, 12, {0, {u32=37487032, u64=37487032}}) = 0
epoll_wait(3, ^CProcess 3826 detached
 <detached ...>

```

```
syslog-ng 3826 root   11r  a_inode                0,9         0    7269 inotify
syslog-ng 3826 root   12u     IPv4            1087399       0t0     UDP rt2m09617.sqa.tbc:48464->10.101.0.55:syslog
```

when I rmmod hello.ko ，the loop of setsockopt(11, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket) disppear. . I am sure not only kernel log messages , but also auth log messages can make this happen. 

```
[root@rt2m09617.sqa.tbc /home/ahao.mah/main]
#rmmod hello
```

## Configuration

```
#cat /etc/syslog-ng/syslog-ng.conf
@version:3.6
@include "scl.conf"
@define allow-config-dups 1
options {
	flush_lines (0);
	threaded(no);
	time_reopen (10);
	log_fifo_size (10000);
	chain_hostnames (no);
	use_dns (no);
	use_fqdn (yes);
	keep_hostname (yes);
	keep_timestamp(no);
	create_dirs (yes);
	dir_perm(0755);
	owner("root");
	group("adm");
	perm(0640);
	stats_freq(0);
};

#syslog-ng client configure,log source is only from local
source s_sys {
    system();
    internal();
};

include /etc/syslog-ng/conf.d;
include /etc/syslog-ng/remote_server.conf;

destination d_sys_cons      { file("/dev/console"); };
destination d_sys_mesg      { file("/var/log/messages"); };
destination d_sys_auth      { file("/var/log/secure"); };
destination d_sys_mail      { file("/var/log/maillog" flush_lines(10)); };
destination d_sys_spol      { file("/var/log/spooler"); };
destination d_sys_boot      { file("/var/log/boot.log"); };
destination d_sys_cron      { file("/var/log/cron"); };
destination d_sys_kern      { file("/var/log/kern"); };
destination d_sys_mlal      { file("/dev/tty0"); };

filter f_sys_snoopy         { program("snoopy"); };
filter f_sys_netstat        { program("netstat"); };
filter f_sys_app            { facility(local0,local1,local2,local3,local4); };
filter f_sys_auth           { facility(auth,authpriv); };
filter f_sys_kernel         { facility(kern); };
filter f_sys_mail           { facility(mail); };
filter f_sys_boot           { facility(local7); };
filter f_sys_cron           { facility(cron); };
filter f_sys_emergency      { level(emerg); };
filter f_sys_news           { facility(uucp) or (facility(news) and level(crit..emerg)); };
filter f_sys_message        { level(info..emerg); };

#anycast_address
#define in /etc/syslog-ng/remote_server.conf
#test_dev: 10.101.0.55
#aliyun: 10.143.22.55
#main_site: 10.195.29.55
#alipay: 10.137.100.55
#destination d_sys_loghost   { udp("\${any_syslog_ip}" port(514) suppress(10)); };

log { source(s_sys); filter(f_sys_snoopy);      flags(final); };
log { source(s_sys); filter(f_sys_app);         flags(final); };
log { source(s_sys); destination(d_sys_loghost); };
log { source(s_sys); filter(f_sys_netstat);     flags(final); };
log { source(s_sys); filter(f_sys_auth);        destination(d_sys_auth);};
log { source(s_sys); filter(f_sys_kernel);      destination(d_sys_kern);};
log { source(s_sys); filter(f_sys_mail);        destination(d_sys_mail);};
log { source(s_sys); filter(f_sys_cron);        destination(d_sys_cron);};
log { source(s_sys); filter(f_sys_emergency);   destination(d_sys_mlal);};
log { source(s_sys); filter(f_sys_boot);        destination(d_sys_boot);};
log { source(s_sys); filter(f_sys_news);        destination(d_sys_spol);};
log { source(s_sys); filter(f_sys_message);     destination(d_sys_mesg); flags(final);};

```

**Comment by czanik** [comment]
I can't say anything about the code part. On the other hand I see, that you use a quite old version. I'd recommend updating to the recently released 3.9, which includes many bugfixes.
My git HEAD repo for Fedora / RHEL 7 currently hosts 3.9.1: https://www.balabit.com/blog/rpm-packages-from-syslog-ng-git-head/

**Comment by furiel** [comment]
I just tried with 3.19, and I see these setsockopt problems even without the kernel module.

Interestingly, the problem occurs even with an "empty" config:
```
@version: 3.19
log {};
```

```
$ timeout 1 strace -e trace=setsockopt ./syslog-ng -Fe -f ../etc/tmp.conf 
syslog-ng: Error setting capabilities, capability management disabled; error='Operation not permitted'
setsockopt(5, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
setsockopt(6, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
setsockopt(7, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
setsockopt(8, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
setsockopt(9, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
setsockopt(11, SOL_SOCKET, SO_OOBINLINE, [1], 4) = 0
[2019-02-08T15:53:15.732672] syslog-ng starting up; version='3.19.1'
strace: Process 12221 detached
[2019-02-08T15:53:16.668455] syslog-ng shutting down; version='3.19.1'
```
With the empty configuration, these errors do not accumulate.

However with a simple file destination:
```
@version: 3.19
log {
    source { example-msg-generator(freq(0.1)); };
    destination { file(/dev/stdout); };
};
```

```
$ timeout 1 strace -e trace=setsockopt ./syslog-ng -Fe -f ../etc/tmp.conf
syslog-ng: Error setting capabilities, capability management disabled; error='Operation not permitted'
setsockopt(5, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
setsockopt(6, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
setsockopt(7, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
setsockopt(8, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
setsockopt(9, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
setsockopt(11, SOL_SOCKET, SO_OOBINLINE, [1], 4) = 0
[2019-02-08T15:57:32.011213] syslog-ng starting up; version='3.19.1'
setsockopt(12, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
Feb  8 15:57:32 furiel -- Generated message. --
setsockopt(12, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
Feb  8 15:57:32 furiel -- Generated message. --
setsockopt(12, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
Feb  8 15:57:32 furiel -- Generated message. --
setsockopt(12, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
Feb  8 15:57:32 furiel -- Generated message. --
setsockopt(12, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
Feb  8 15:57:32 furiel -- Generated message. --
setsockopt(12, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
Feb  8 15:57:32 furiel -- Generated message. --
setsockopt(12, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
Feb  8 15:57:32 furiel -- Generated message. --
setsockopt(12, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
Feb  8 15:57:32 furiel -- Generated message. --
setsockopt(12, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
Feb  8 15:57:32 furiel -- Generated message. --
setsockopt(12, SOL_SOCKET, SO_OOBINLINE, [1], 4) = -1 ENOTSOCK (Socket operation on non-socket)
strace: Process 12337 detached
[2019-02-08T15:57:32.948192] syslog-ng shutting down; version='3.19.1'
Feb  8 15:57:32 furiel -- Generated message. --
$
```
the error occurs per message.

This `SO_OOBINLINE` call does not show up in syslog-ng code, however it is called in [ivykis](https://github.com/buytenh/ivykis/blob/8f8eb788f96ed3a128d240370f1c6afd3c5d1748/src/iv_fd.c#L342).

I will check if I can reproduce the problem without syslog-ng, only with ivykis.

**Comment by bazsi** [comment]
it is done in iv_fd_register(), and our logreader unfortunately
registers/unregisters fds at every poll iteration, which this PR is aiming
at to address.

https://github.com/balabit/syslog-ng/pull/2531

**Comment by gaborznagy** [comment]
Fixed by #2531.
